avm/ptn/sa/conversation-knowledge-mining/main.bicep

```
@@ -0,0 +1,1395 @@
+  name: format(localModuleDeploymentNameFormat, 'rbac-aiproject-aideveloper')
+  params: {
+    aiServicesProjectResourceName: moduleAIFoundry.outputs.aiProjectName
+    ragFunctionPrincipalId: moduleFunctionRAG.outputs.principalId
+    ragFunctionResourceId: moduleFunctionRAG.outputs.resourceId
+  }
+}
+
+//========== CKM Webapp ========== //
+
+module avmServerFarmWebapp 'br/public:avm/res/web/serverfarm:0.4.1' = {
+  name: format(avmDeploymentNameFormat, webAppServerFarmResourceName)
+  params: {
+    name: webAppServerFarmResourceName
+    tags: tags
+    location: webAppServerFarmLocation
+    enableTelemetry: enableTelemetry
+    diagnosticSettings: [{ workspaceResourceId: avmLogAnalyticsWorkspace.outputs.resourceId }]
+    skuName: webAppServerFarmSkuName
+    kind: 'linux'
+  }
+}
+
+module moduleWebsiteWebapp 'modules/webapp.bicep' = {
+  name: format(localModuleDeploymentNameFormat, 'webapp')
+  params: {
+    deploymentName: format(avmDeploymentNameFormat, webAppResourceName)
+    webAppName: webAppResourceName
+    tags: tags
+    location: webAppLocation
+    enableTelemetry: enableTelemetry
+    serverFarmResourceId: avmServerFarmWebapp.outputs.resourceId
+    appInsightsResourceId: moduleAIFoundry.outputs.applicationInsightsResourceId
+    aiFoundryAIServicesName: moduleAIFoundry.outputs.aiServicesName
+    webAppImageName: 'DOCKER|${webAppDockerImageContainerRegistryUrl}/${webAppDockerImageName}:${webAppDockerImageTag}'
+    appInsightsInstrumentationKey: moduleAIFoundry.outputs.applicationInsightsInstrumentationKey
+    aiServicesEndpoint: moduleAIFoundry.outputs.aiServicesEndpoint
+    gptModelName: aiFoundryAIServicesGptModelName
+    gptModelVersionPreview: varAiFoundryAiServiceGPTModelVersionPreview
+    aiServicesResourceName: moduleAIFoundry.outputs.aiServicesName
+    chartsFunctionDefaultHostName: functionCharts.properties.defaultHostName
+    //chartsFunctionDefaultHostName: avmFunctionCharts.outputs.defaultHostname
+    chartsFunctionFunctionName: functionChartsFunctionName
+    webAppAppConfigReact: varWebAppAppConfigReact
+    cosmosDbSqlDbName: varCosmosDbSqlDbName
+    cosmosDbSqlDbNameCollectionName: varCosmosDbSqlDbCollName
+    ragFunctionDefaultHostName: moduleFunctionRAG.outputs.defaultHostName
+    ragFunctionFunctionName: functionRagFunctionName
+    cosmosDbResourceName: avmCosmosDB.outputs.name
+  }
+}
+
+module rbac 'modules/rbac-cosmosdb-contributor.bicep' = {
+  name: format(localModuleDeploymentNameFormat, 'rbac-cosmosdb-contributor')
+  params: {
+    cosmosDBAccountName: avmCosmosDB.outputs.name
+    principalId: moduleWebsiteWebapp.outputs.webAppSystemAssignedPrincipalId
+  }
+}
+
+@description('The resource group the resources were deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The url of the webapp where the deployed Conversation Knowledge Mining solution can be accessed.')
+output webAppUrl string = '${webAppResourceName}.azurewebsites.net'
+
+@export()
```

```suggestion

// =============== //
//   Definitions   //
// =============== //

@export()
```

Not 'necessary' but a nice seperator for orientation